Source/core/Proxy.h

```diff
@@ -185,17 +185,15 @@ namespace Core {
         // -----------------------------------------------------
         HAS_MEMBER(Initialize, hasInitialize);
 
-        typedef hasInitialize<CONTEXT, uint32_t (CONTEXT::*)()> TraitInitialize;
-
-        template <typename TYPE>
-        inline typename Core::TypeTraits::enable_if<ProxyService<TYPE>::TraitInitialize::value, uint32_t>::type
+        template <typename TYPE=CONTEXT>
+        inline typename Core::TypeTraits::enable_if<hasInitialize<TYPE, uint32_t (TYPE::*)()>::value, uint32_t>::type
         __Initialize()
```

Comment:
Please also check if the usage of these methods now should not be changed as well (template params removed). e.g. line 58, 99, 103 etc.)